### PR TITLE
Yaml linting files

### PR DESCRIPTION
**Pull request by achinthagunasekara**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind

cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Running YAML linting on YAML files in `cmd/kubeadm/app` and fixing issues

**Which issue(s) this PR fixes**:

N/A

Fixes #

N/A

**Special notes for your reviewer**:

YAML linting files

**Does this PR introduce a user-facing change?**:

No

```release-note

```

### CODE REVIEW

**Review by yagonobre — COMMENTED** [review]
Thanks @achinthagunasekara
/ok-to-test
/priority backlog
/kind cleanup
/release-note-none

File: cmd/kubeadm/app/cmd/upgrade/testdata/diff_controlplane_config.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+---
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/upgrade/testdata/diff_controlplane_config.yaml]: hm, what type of linter is that?
files with a single kind should no have a leading `---`.
/hold

File: cmd/kubeadm/app/util/config/testdata/conversion/controlplane/internal.yaml [hunk 2]
```diff
@@ -1,29 +1,30 @@
+---
 APIServer:
   CertSANs: null
   ExtraArgs:
     authorization-mode: Node,RBAC,Webhook
   ExtraVolumes:
-  - HostPath: /host/read-only
```
Comment (neolit123) [inline cmd/kubeadm/app/util/config/testdata/conversion/controlplane/internal.yaml]: i don't think we should do these indentations.

**Review by neolit123 — COMMENTED** [review]
thank you for the PR @achinthagunasekara 
can you explain the linter your are using?
some of these changes seem redundant.

/kind cleanup
/priority awaiting-more-evidence
@kubernetes/sig-cluster-lifecycle-pr-reviews

**Review by rosti — COMMENTED** [review]
Thanks for the contribution @achinthagunasekara !

I think, that @neolit123 is quite right on all counts here. This linter and the Kubernetes YAML serializer seem to have different opinions on what a YAML file should look like. Also, I find it rather unnecessary to have each file start with a `---`.

If you look forward to contributing to kubeadm development, please get in touch with us on slack (#sig-cluster-lifecycte channel on k8s slack), come to the [kubeadm office hours meetings](https://docs.google.com/document/d/130_kiXjG7graFNSnIAgtMS1G8zPDwpkshgfRYS0nggo/edit) and/or pick an issue from kubeadm's [issue tracker](https://github.com/kubernetes/kubeadm/issues) (`good first issue` or `help wanted` are good places to start).

/hold

**Comment by achinthagunasekara** [comment]
I'm using https://pypi.org/project/yamllint/ for linting YAML files

File: cmd/kubeadm/app/cmd/upgrade/testdata/diff_controlplane_config.yaml (hunk 1 above)
Comment (achinthagunasekara) [reply cmd/kubeadm/app/cmd/upgrade/testdata/diff_controlplane_config.yaml]: I'm using https://pypi.org/project/yamllint/ :)

**Comment by achinthagunasekara** [comment]
Thanks @rosti I've joined the slack channel :)

**Comment by achinthagunasekara** [comment]
Thanks @rosti I've joined the slack channel :)

**Pull request closed without merging**